servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpRequestMetaData.java

```diff
@@ -433,6 +433,33 @@ private void query(final Map<String, List<String>> params) {
         requestTarget(sb.toString());
     }
 
+    @Nullable
+    @Override
+    public String fragment() {
+      return lazyParseRequestTarget().fragment();
+    }
+
+    @Override
+    public HttpRequestMetaData fragment(@Nullable String fragment) {
+
+      if (fragment == null && fragment() == null) {
+        return this;
+      }
+
+      String originalRequestTarget = requestTarget();
+
+      if (lazyParseRequestTarget().fragment() != null) {
+        int fragmentLength = lazyParseRequestTarget().fragment().length() + 1;
```

Comment:
The short equivalent of `lazyParseRequestTarget().fragment()` will be `fragment()`. Looks like we need it for all code paths, consider caching in a local variable at the top of the method.